cmd/pbstatuser/main.go

```diff
@@ -20,25 +30,162 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+const ChannelBuffer = 32
+
+type SourceInfo struct {
+	Authentication clients.Authentication `json:"-"`
+
+	SourceID string `json:"resource_id"`
+
+	Identity identity.XRHID `json:"-"`
+}
+
+var (
+	chAws        = make(chan SourceInfo, ChannelBuffer)
+	chAzure      = make(chan SourceInfo, ChannelBuffer)
+	chGcp        = make(chan SourceInfo, ChannelBuffer)
+	chSend       = make(chan kafka.SourceResult, ChannelBuffer)
+	receiverWG   = sync.WaitGroup{}
+	processingWG = sync.WaitGroup{}
+	senderWG     = sync.WaitGroup{}
+)
+
 func processMessage(ctx context.Context, message *kafka.GenericMessage) {
 	logger := ctxval.Logger(ctx)
+
 	// This is just a test to see if Kafka is configured properly in ephemeral
 	logger.Warn().Msg("RECEIVED KAFKA MESSAGE")
 
-	// TODO: implement the source checking for AWS, Azure, GCP
+	// Get source id
+	asm, err := kafka.NewAvailabilityStatusMessage(message)
+	if err != nil {
+		logger.Warn().Msgf("could not get availability status message %s", err)
```

Comment:
We do log messages with capitals, errors with lower case.

```suggestion
		logger.Warn().Msgf("Could not get availability status message %s", err)
```

There's a lot more down below. It's like 90% of the logs in this format, let's try to keep that number high